### PR TITLE
feat(TPG >= 4.59)!: added preconfigured_waf_config block

### LINKED ISSUE #33: Add preconfigured_waf_config block will be added to the module?
Hi,
Is there a plan to add [preconfigured_waf_config](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_security_policy#preconfigured_waf_config) block to the module?

### PR DESCRIPTION
**Pull request by imrannayer**
Fix #33

### CODE REVIEW

File: docs/upgrading_to_v1.0.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Upgrading to v1.0.0
```
Comment (bharathkkb) [inline docs/upgrading_to_v1.0.md:1]: For pre 1.0 major releases, you can continue to use 0.x tags. If you think you are ready to tag 1.0 feel free to go ahead but thought I would call out that it is not a requirement.

File: variables.tf [hunk 2]
```diff
@@ -63,13 +63,36 @@ variable "pre_configured_rules" {
...
       header_name  = optional(string)
       header_value = optional(string)
     })), [])
+
+    preconfigured_waf_config_exclusion = optional(object({
+      target_rule_set = string
+      target_rule_ids = optional(list(string), [])
+      request_header = optional(list(object({
+        operator = string
+        value    = optional(string, "test")
```
Comment (bharathkkb) [inline variables.tf:78]: Would users actually want to use a "test" default value?

File: docs/upgrading_to_v1.0.md (hunk 1 above)
Comment (imrannayer) [reply docs/upgrading_to_v1.0.md:1]: With module covering almost all the available features and used by many customers we can move it to 1.0.

File: variables.tf (hunk 2 above)
Comment (imrannayer) [reply variables.tf:78]: Fixed. I added it to test something.

**Pull request merged**